packages/synapse-react-client/src/components/ChallengeRequirementsModal/ChallengeRequirementsModal.tsx

```diff
@@ -0,0 +1,120 @@
+ *  2. If the user has accepted the requirements, the user can click 'Register' to join the team
+ *     If the user has not accepted the requirements, the user cannot register for the challenge.
+ * @constructor
+ */
+export default function ChallengeRequirementsModal(
+  props: ChallengeRequirementsModalProps,
+) {
+  const { open, projectId, onRegisterComplete, onCancel } = props
+  const { data: project } = useGetEntity(projectId)
+
+  const { accessToken } = useSynapseContext()
+
+  const { data: challenge } = useGetEntityChallenge(projectId)
+  const isLoggedIn = !!accessToken
+  const { data: userProfile } = useGetCurrentUserProfile({
+    enabled: isLoggedIn,
+  })
+
+  const participantTeamId = challenge?.participantTeamId
+  const userId = userProfile?.ownerId
+
+  // Verify that user is a member of the participant team
+  const { data: teamMembershipStatus } = useGetMembershipStatus(
+    participantTeamId!,
+    userId!,
+    {
+      enabled: Boolean(participantTeamId && userId),
+    },
+  )
+
+  const { mutateAsync: addTeamMember } = useAddMemberToTeam()
+
+  const {
+    data: fetchedRequirementsForTeam,
+    isSuccess: hasFetchedRequirements,
+  } = useGetAccessRequirementsForTeam(challenge?.participantTeamId!, {
+    enabled: Boolean(challenge?.participantTeamId),
+  })
+
+  const requirementStatusesQueryResults = useGetAccessRequirementStatuses(
+    fetchedRequirementsForTeam?.map(ar => String(ar.id)) ?? [],
+  )
+
+  const meetsAllRequirements =
+    hasFetchedRequirements &&
+    requirementStatusesQueryResults.every(queryResult => {
+      return queryResult.status === 'success' && queryResult.data.isApproved
+    })
+
+  const registerForChallenge = useCallback(async () => {
+    if (participantTeamId && userId && !teamMembershipStatus?.isMember) {
+      await addTeamMember({
+        teamId: participantTeamId,
+        userId,
+      })
+      onRegisterComplete()
+    }
+  }, [
+    addTeamMember,
+    onRegisterComplete,
+    participantTeamId,
+    teamMembershipStatus?.isMember,
+    userId,
+  ])
+
+  return (
+    <ConfirmationDialog
+      open={open}
+      title={`Challenge Terms and Conditions`}
+      content={
+        participantTeamId && (
+          <AccessRequirementList
+            subjectId={participantTeamId}
+            subjectType={RestrictableObjectType.TEAM}
+            teamId={challenge?.participantTeamId}
+            renderAsModal={false}
+            requestObjectName={project?.name}
+          />
+        )
+      }
+      onCancel={onCancel}
+      onConfirm={() => {
+        registerForChallenge()
+      }}
+      confirmButtonProps={{
+        children: teamMembershipStatus?.isMember ? 'Continue' : 'Register',
+        // It's possible that the user joined the team, but requirements on the team changed since they became a member
+        // They do not need to re-join the team, but they should still be prompted to accept the requirements
+        disabled: !meetsAllRequirements,
```

Comment:
I do not think it's possible to show this dialog if you're in this state, but this case is handled if that ever changes.